cmd_actuators_mux/src/cmd_actuator_mux_nodelet.cpp

```diff
@@ -0,0 +1,418 @@
+    set_rpm_output_topic_pub =
+        nh.advertise<thruster_control::SetRPM>(set_rpm_output_topic_name, 10);
+    NODELET_DEBUG_STREAM("Subscribe to output topic '" << output_name << "'");
+  }
+  else
+  {
+    NODELET_DEBUG_STREAM("No need to re-subscribe to output topic '" << output_name << "'");
+  }
+
+  /******** Input Subscribers *******************************/
+  try
+  {
+    cmd_set_rpm_subs.configure(doc["set_rpm_subscribers"]);
+  }
+  catch (EmptyCfgException& e)
+  {
+    NODELET_WARN_STREAM(
+        "yaml configured zero subscribers, check yaml content: set_rpm_subscribers");
+  }
+  catch (YamlException& e)
+  {
+    NODELET_ERROR_STREAM("set_rpm_subscribers: yaml parsing problem [" << std::string(e.what())
+                                                                       << "]");
+  }
+
+  double set_rpm_longest_timeout = 0.0;
+  for (unsigned int i = 0; i < cmd_set_rpm_subs.size(); i++)
+  {
+    if (!cmd_set_rpm_subs[i]->subs)
+    {
+      cmd_set_rpm_subs[i]->subs = nh.subscribe<thruster_control::SetRPM>(
+          cmd_set_rpm_subs[i]->topic, 10, CmdSetRPMFunctor(i, this));
+      NODELET_DEBUG("Thruster Set RPM Mux : subscribed to '%s' on topic '%s'. pr: %d, to: %.2f",
+                    cmd_set_rpm_subs[i]->name.c_str(), cmd_set_rpm_subs[i]->topic.c_str(),
+                    cmd_set_rpm_subs[i]->priority, cmd_set_rpm_subs[i]->timeout);
+    }
+    else
+    {
+      NODELET_DEBUG_STREAM("Thruster Set RPM Mux : no need to re-subscribe to input topic '"
+                           << cmd_set_rpm_subs[i]->topic << "'");
+    }
+
+    if (!cmd_set_rpm_subs[i]->timer)
+    {
+      // Create (stopped by now) a one-shot timer for every subscriber, if it doesn't exist yet
+      cmd_set_rpm_subs[i]->timer = pnh.createTimer(ros::Duration(cmd_set_rpm_subs[i]->timeout),
+                                                   SetRPMTimerFunctor(i, this), true, false);
+    }
+
+    if (cmd_set_rpm_subs[i]->timeout > set_rpm_longest_timeout)
+      set_rpm_longest_timeout = cmd_set_rpm_subs[i]->timeout;
+  }
+
+  if (!set_rpm_common_timer)
+  {
+    // Create another timer for  messages from any source, so we can
+    // dislodge last active source if it gets stuck without further messages
+    set_rpm_common_timer_period = longest_timeout * 2.0;
+    set_rpm_common_timer = pnh.createTimer(ros::Duration(set_rpm_common_timer_period),
+                                           SetRPMTimerFunctor(GLOBAL_TIMER, this), true, false);
+  }
+  else if (longest_timeout != (set_rpm_common_timer_period / 2.0))
+  {
+    // Longest timeout changed; just update existing timer period
+    set_rpm_common_timer_period = longest_timeout * 2.0;
+    set_rpm_common_timer.setPeriod(ros::Duration(set_rpm_common_timer_period));
+  }
+
+  // check that the subscribers input have the same name in YAML
+  for (int i = 0; i < cmd_set_rpm_subs.size(); i++)
+  {
+    ROS_ASSERT_MSG(!cmd_set_rpm_subs[i]->name.compare(cmd_fin_angles_subs[i]->name),
```

Comment:
@patapufate we have to assert that both arrays have the same length as well.

Reply:
added in  5f20933